src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -0,0 +1,529 @@
+                    return false;
+                }
+            }
+
+            result = new AssemblyNameParts(name, version, cultureName, flags, pkt);
+            return true;
+        }
+
+        private static bool IsAttribute(string candidate, string attributeKind)
+            => candidate.Equals(attributeKind, StringComparison.OrdinalIgnoreCase);
+
+        private static bool TryParseVersion(string attributeValue, ref Version? version)
+        {
+#if NET8_0_OR_GREATER
+            ReadOnlySpan<char> attributeValueSpan = attributeValue;
+            Span<Range> parts = stackalloc Range[5];
+            parts = parts.Slice(0, attributeValueSpan.Split(parts, '.'));
+#else
+            string[] parts = attributeValue.Split('.');
+#endif
+            if (parts.Length is < 2 or > 4)
+            {
+                return false;
+            }
+
+            Span<ushort> versionNumbers = stackalloc ushort[4];
+            for (int i = 0; i < versionNumbers.Length; i++)
+            {
+                if ((uint)i >= (uint)parts.Length)
+                {
+                    versionNumbers[i] = ushort.MaxValue;
+                    break;
+                }
+
+                if (!ushort.TryParse(
+#if NET8_0_OR_GREATER
+                    attributeValueSpan[parts[i]],
+#else
+                    parts[i],
+#endif
+                    NumberStyles.None, NumberFormatInfo.InvariantInfo, out versionNumbers[i]))
+                {
+                    return false;
+                }
+            }
+
+            if (versionNumbers[0] == ushort.MaxValue ||
+                versionNumbers[1] == ushort.MaxValue)
+            {
+                return false;
+            }
+
+            version =
+                versionNumbers[2] == ushort.MaxValue ? new Version(versionNumbers[0], versionNumbers[1]) :
+                versionNumbers[3] == ushort.MaxValue ? new Version(versionNumbers[0], versionNumbers[1], versionNumbers[2]) :
+                new Version(versionNumbers[0], versionNumbers[1], versionNumbers[2], versionNumbers[3]);
+
+            return true;
+        }
+
+        private static bool TryParseCulture(string attributeValue, out string? result)
+        {
+            if (attributeValue.Equals("Neutral", StringComparison.OrdinalIgnoreCase))
+            {
+                result = "";
+                return true;
+            }
+
+            result = attributeValue;
+            return true;
+        }
+
+        private static bool TryParsePKT(string attributeValue, bool isToken, ref byte[]? result)
+        {
+            if (attributeValue.Equals("null", StringComparison.OrdinalIgnoreCase) || attributeValue == string.Empty)
+            {
+                result = Array.Empty<byte>();
+                return true;
+            }
+
+            if (isToken && attributeValue.Length != 8 * 2)
+            {
+                return false;
+            }
+
+            byte[] pkt = new byte[attributeValue.Length / 2];
```

Comment:
What happens to the leftover char if _attributeValue_ has an odd length?

Reply:
Another great catch! (that was with .NET forever until now ;) )